pkg/testutil/firestore_mock.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package testutil provides utilities that are intended to enable easier
+// and more concise writing of unit test code.
+package testutil
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+
+	"cloud.google.com/go/firestore"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	emptypb "github.com/golang/protobuf/ptypes/empty"
+	"google.golang.org/api/option"
+	firestorepb "google.golang.org/genproto/googleapis/firestore/v1"
+	"google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+)
+
+var (
+	_ = io.EOF
+	_ = ptypes.MarshalAny
+	_ status.Status
+)
+
+type MockFirestoreServer struct {
+	// Embed for forward compatibility.
+	// Tests will keep working if more methods are added
+	// in the future.
+	firestorepb.UnimplementedFirestoreServer
+
+	Reqs []proto.Message
+
+	// If set, all calls return this error.
+	Error error
+
+	// responses to return if err == nil
+	Resps []proto.Message
+}
+
+func (s *MockFirestoreServer) GetDocument(ctx context.Context, req *firestorepb.GetDocumentRequest) (*firestorepb.Document, error) {
+	md, _ := metadata.FromIncomingContext(ctx)
+	if xg := md["x-goog-api-client"]; len(xg) == 0 || !strings.Contains(xg[0], "gl-go/") {
```

Comment:
the code borrowed from here https://github.com/googleapis/google-cloud-go/blob/6d1e9fe399d0f9adf1a80107be1b27c725b02571/firestore/apiv1/mock_test.go#L63.
I tried to investigate why we need such metadata validation here, but can't find the clues.

Reply:
Seems to be ensuring that the firestore client is identifying itself. Probably unnecessary for us. 

https://cloud.google.com/apis/docs/system-parameters